great_expectations/datasource/fluent/sources.py

```diff
@@ -475,8 +475,6 @@ def add_datasource(
             )
             logger.debug(f"Adding {datasource_type} with {datasource.name}")
             datasource._data_context = self._data_context
-            # config provider needed for config substitution
-            datasource._config_provider = self._data_context.config_provider
```

Comment:
No reason for this to be set separately, just one more point of failure.
We can pull the `config_provider` from the `data_context` directly